text/0000-grid-primitives.md

```diff
@@ -0,0 +1,618 @@
+- Feature Name: Primitives
+- Start Date: TBD
+- RFC PR: TBD
+
+# Summary
+[summary]: #summary
+
+Grid Primitives for Hyperledger Grid provides a reusable, standard approach to
+defining, storing, and consuming properties within smart contracts, software
+libraries, and network-based APIs.
+
+# Motivation
+[motivation]: #motivation
+
+Several components within Grid will store and retrieve properties which are
+defined at runtime. To properly store and validate these properties, we need
+property definitions which minimally include the property’s type (integer,
+string, enum, etc.). In addition, the properties (for example, product
+description, GPS location, or product dimensions) should always be stored and
+exchanged using the same format within Grid components.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## PropertyDefinition
+
+A property is defined using a `PropertyDefinition` which includes the following:
+
+- Data type (one of: BYTES, BOOLEAN, NUMBER, STRING, ENUM, STRUCT, LOCATION)
```

Comment:
LOCATION is not called out below.

Reply:
I think I missed that - was thinking that it could just be handled as a struct, going forward.  Probably more efficient to have a location as a top-level data type.